src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -120,6 +120,20 @@ public boolean containsEntryWithId(String id) {
         return internalIDs.contains(id);
     }
 
+    /**
+     * Returns whether an entry with the given bibtex key exists.
+     */
+    public synchronized boolean containsEntryWithKey(String key) {
+        int keyHash = key.hashCode(); // key hash for better performance
```

Comment:
Two-stage solution. Removed `containsEntryWithKey` here and will do the other changes in #1568.